### PR TITLE
ui-speedspacechart: fix pr names display

### LINKED ISSUE #289: ui-speedspacechart: fix pr's names display
### Description and goal

actually the pr's names are overlapping, they should follow the mockup design and don't overlap in convenience with the zoom level

### Acceptance criteria

- [ ] the pr's names follow the mockup design
- [ ] some aren't displayed if the zoom level isn't sufficient

### PR DESCRIPTION
**Pull request by Yohh**
close #289

### CODE REVIEW

**Comment by Math-R** [comment]
Sorry miss navigation and fail click, need to review....

File: ui-speedspacechart/src/components/helpers/drawElements/gridX.ts [hunk 1]
```diff
@@ -7,6 +7,29 @@ const { MARGIN_LEFT, MARGIN_RIGHT, MARGIN_TOP, MARGIN_BOTTOM, CURVE_MARGIN_SIDES
...
+      ((width - CURVE_MARGIN_SIDES) / maxPositionValues(store).maxPosition) *
+      ratioX +
+    CURVE_MARGIN_SIDES / 2;
+
+  const filteredStops = stops.filter(({ position }) => {
+    const actualX =
+      position.start *
+        ((width - CURVE_MARGIN_SIDES) / maxPositionValues(store).maxPosition) *
+        ratioX +
+      CURVE_MARGIN_SIDES / 2;
```
Comment (Uriel-Sautron) [inline ui-speedspacechart/src/components/helpers/drawElements/gridX.ts]:
```suggestion
const getStopPosition = (position: LayerData<string>['position']) =>
    position.start *
      ((width - CURVE_MARGIN_SIDES) / maxPositionValues(store).maxPosition) *
      ratioX +
    CURVE_MARGIN_SIDES / 2;

  let stopPosition = getStopPosition(stops[0].position);

  const filteredStops = stops.filter(({ position }) => {
    const actualX = getStopPosition(position);
```

File: ui-speedspacechart/src/components/helpers/drawElements/step.ts [hunk 2]
```diff
@@ -4,9 +4,36 @@ import { clearCanvas, maxPositionValues } from '../../utils';
...
+const STEP_RADIUS = 0.3;
+
 export const drawStep = ({ ctx, width, height, store }: DrawFunctionParams) => {
   const { stops, ratioX, leftOffset } = store;
 
+  if (stops.length === 0) {
+    return;
+  }
+
+  let stopPosition =
```
Comment (Uriel-Sautron) [inline ui-speedspacechart/src/components/helpers/drawElements/step.ts]: You can export `getStopPosition` from gridX.ts (or add it to utils) and use it here

**Review by Uriel-Sautron — COMMENTED** [review]
Nice rendering, thank you !!! Tested.
Just a few comments.

File: ui-speedspacechart/src/components/helpers/drawElements/gridX.ts (hunk 1 above)
Comment (Yohh) [reply ui-speedspacechart/src/components/helpers/drawElements/gridX.ts]: done

File: ui-speedspacechart/src/components/helpers/drawElements/step.ts (hunk 2 above)
Comment (Yohh) [reply ui-speedspacechart/src/components/helpers/drawElements/step.ts]: done

**Review by Uriel-Sautron — APPROVED** [review]
LGTM !!

**Review by Math-R — APPROVED** [review]
lgtm & tested

**Pull request merged**